From a09986f4272f0d64f1ea93227399ac7b50cb0a11 Mon Sep 17 00:00:00 2001
From: TraHT <traht@fsoft.com.vn>
Date: Thu, 28 Jun 2018 18:50:25 +0700
Subject: Modify
 1005-Fixed-memory-corruption-when-focus-listener-is-moved.patch to fix hunk

Signed-off-by: TraHT <traht@fsoft.com.vn>
---
 desktop-shell/shell.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index f80088f..d691972 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -726,9 +726,12 @@ focus_state_set_focus(struct focus_state *state,
 	}
 
 	state->keyboard_focus = surface;
-	if (surface)
+	if (surface) {
+        /* Remove link references properly before moving to other signal, other surface still could keep reference to it */
+		wl_list_remove(&state->surface_destroy_listener.link);
 		wl_signal_add(&surface->destroy_signal,
 			      &state->surface_destroy_listener);
+	}
 }
 
 static void
-- 
2.7.4

